pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -0,0 +1,184 @@
+import io
+import os
+from pytorch_lightning.core.lightning import LightningModule
+import torch
+from pytorch_lightning.utilities.distributed import rank_zero_only
+from pytorch_lightning.utilities.seed import seed_everything
+from typing import Any, Dict, Optional
+from pytorch_lightning.plugins.training_type.ddp_spawn import DDPSpawnPlugin
+from pytorch_lightning.utilities import _TPU_AVAILABLE, rank_zero_warn,
+
+from pytorch_lightning.utilities.apply_func import move_data_to_device
+
+if _TPU_AVAILABLE:
+    import torch_xla
+    import torch_xla.core.xla_model as xm
+    import torch_xla.distributed.parallel_loader as xla_pl
+    import torch_xla.distributed.xla_multiprocessing as xmp
+
+class TPUSpawnPlugin(DDPSpawnPlugin):
+    def __init__(self, parallel_devices, num_nodes=1, **kwargs: Dict[str, Any]):
+
+        parallel_devices = [xm.xla_device(device) if isinstance(device, int) else device for device in parallel_devices]
```

Comment:
We don't need to specify the device when using spawn. calling `xm.xla_device()` within `xmp.spawn` should automatically assign a core to each process.

Reply:
But does it hurt? Since this is the unified interfaces for all the plugins...

Reply:
When you call `xm.xla_device(device)` outside of spawn, the first call will attach that device to the main process and results in this error when spawn is called after it: https://github.com/PyTorchLightning/pytorch-lightning/issues/3169